WordPress/gutenberg
# Navigator: use CSS animations instead of framer-motion

## What?


Related to #55892

Use vanilla CSS animations for the `Navigator` components instead of `framer-motion`

## Why?


**This PR doesn't fix the real cause for #55892, but aims at improving a visual side-effect caused by it**

`framer-motion` animations updated the `style` attribute of the animating DOM element, setting a new value for every frame. In other words, `framer-motion` performed some calculations in every animation frame to interpolate the exact values of the animation (ie. translation). These calculations happen on the main thread, and therefore the animation can get quite janky if the main thread is busy with other tasks (see #55892).

Using vanilla CSS allows the animation to [run directly on the GPU](https://www.smashingmagazine.com/2016/12/gpu-animation-doing-it-right/), running smoothly regardless of how busy the CPU is.

Furthermore, the animations required for this component are simple enough to be written with vanilla CSS, and don't necessarily require the sophisticated orchestration features that `framer-motion` offers.

## How?


- Moved all styles to separate `styles.ts` file
- Replaced `framer-motion` with CSS animations
- added [`contain: strict`](https://developer.mozilla.org/en-US/docs/Web/CSS/contain) to the provider screen wrapper to further potentially enhance rendering performance

Note: with this PR, **navigator screens don't have an exit animation anymore**, as it would require a larger refactor (that's because the previous `NavigatorScreen`'s wrapper element gets removed from the DOM as soon the selected screen `id` changes).

## Testing Instructions





1. Interact with the `Navigator` component in Storybook:
    - Make sure that the first screen doesn't have an enter animation
    - Make sure that all subsequent animations (both backward and forward) animate as expected
    - Make sure that the animations are inverted in contexts with RTL direction
    - Make sure that the animations are disabled when the `prefers-reduced-motion` option is enabled
2. Open the side editor, and interact with the sidebar:
    - Make sure that navigating across submenus works as expected and the animation looks as expected
    - Make sure that the animation doesn't slow down and get janky like on `trunk` (see #55892)
3. Open the global styles sidebar in the site editor:
    - Make sure that the sidebar works and looks as on `trunk`, especially when navigating across the submenus

## Screenshots or screencast 

### Before (`trunk`)

Notice the jankiness (even while testing on a high-end machine), especially while navigating to/from the "Patterns" navigator screen

https://github.com/WordPress/gutenberg/assets/1083581/31eafcd0-60e0-4d66-8127-a4711ae46282



### After (this PR)

No jankiness

https://github.com/WordPress/gutenberg/assets/1083581/7d3593ea-3fa5-4350-9801-078addad3af9
